### PR TITLE
[4.0] Simplify keepalive.js

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- Remove IIFE
- Remove redundant checks
- Switch from XHR to sendBeacon https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon

### Testing Instructions
Apply the PR
Change the `Session Lifetime (minutes)` to `1` in `Global Configuration` -> `System`
Observe the Other in the network tab of your browser's console

### Expected result AFTER applying this Pull Request

Screenshot 2021-10-02 at 13 11 26

### Documentation Changes Required

None, We're switching from XHR to sendBeacon which is widely supported 
https://caniuse.com/beacon
Screenshot 2021-10-02 at 13 25 43

### CODE REVIEW

File: build/media_source/system/js/keepalive.es6.js [hunk 1]
```diff
@@ -11,36 +11,19 @@
  * @package  Joomla.JavaScript
  * @since    3.7.0
  */
-((document, Joomla) => {
-  'use strict';
```
Comment (Fedik) [inline build/media_source/system/js/keepalive.es6.js:15]: why do you remove it? it is not a module

Comment (dgrammatiko) [reply build/media_source/system/js/keepalive.es6.js:15]: You're right, I've fixed it, now it's a module. Also, it means that most of the core scripts are forced to be ES2017 by default and that is wrong

File: build/media_source/system/joomla.asset.json [hunk 2]
```diff
@@ -11,16 +11,28 @@
...
+        "nomodule": true
+      }
+    },
     {
       "name": "keepalive",
       "type": "script",
       "class": "KeepaliveAssetItem",
       "dependencies": [
-        "core"
+        "keepalive.es5"
```
Comment (Fedik) [inline build/media_source/system/joomla.asset.json]: please keep `core` here also, 
when we drop all es5 someone will forget to add it back

Comment (dgrammatiko) [reply build/media_source/system/joomla.asset.json]: done

**Review by Stuartemk — COMMENTED** [review]
IMG_20211002_124634

**Comment by dgrammatiko** [comment]
@Stuartemk yes all these are correct...

Edit: probably you're totally unaware how Joomla is dealing with the static assets, please read the readme.md: https://github.com/joomla/joomla-cms/tree/4.0-dev/build

**Comment by Stuartemk** [comment]
> @Stuartemk yes all these are correct...
> 
> Edit: probably you're totally unaware how Joomla is dealing with the static assets, please read the readme.md: https://github.com/joomla/joomla-cms/tree/4.0-dev/build

Ok
What I saw is the difference that one is es5 and another is es6

I do not understand why, I admit my lack of knowledge, but since I mentioned that and you confirm that it is well I trust.

**Review by ditsuke — APPROVED** [review]

**Comment by Fedik** [comment]
Simplifying turns out not to simple :smile:

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 1652beb4c95d5cac984485f937e4aa9c3200cbbe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35725">issues.joomla.org/tracker/joomla-cms/35725</a>.</sub>

File: build/media_source/system/joomla.asset.json [hunk 3]
```diff
@@ -11,16 +11,29 @@
...
+    {
+      "name": "keepalive.es5",
+      "type": "script",
+      "dependencies": [
+        "core"
+      ],
+      "uri": "system/keepalive-es5.min.js",
+      "attributes": {
+        "defer": true,
+        "nomodule": true
```
Comment (ditsuke) [inline build/media_source/system/joomla.asset.json:23]: I don't fully understand the assets system, why the `"nomodule": true` here but `"type": "module"` for non-es5?

Comment (dgrammatiko) [reply build/media_source/system/joomla.asset.json:23]: https://web.dev/publish-modern-javascript/

Comment (ditsuke) [reply build/media_source/system/joomla.asset.json:23]: Thank you! I have much to learn here

**Comment by ditsuke** [comment]
I have tested this item :white_check_mark: successfully on 1652beb4c95d5cac984485f937e4aa9c3200cbbe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35725">issues.joomla.org/tracker/joomla-cms/35725</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35725">issues.joomla.org/tracker/joomla-cms/35725</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**